src/components/CalendarDay.jsx

```diff
@@ -16,7 +16,7 @@ const propTypes = forbidExtraProps({
   day: momentPropTypes.momentObj,
   daySize: nonNegativeInteger,
   isOutsideDay: PropTypes.bool,
-  modifiers: PropTypes.object,
+  modifiers: PropTypes.instanceOf(Set),
```

Comment:
As long as the PR is already semver-major, and as long as we ensure that these requirements (`Set` and `Array.from` being globally polyfilled) are documented in the readme, I'm a solid 👍 on it.